spark/src/main/scala/org/apache/spark/sql/delta/OptimisticTransaction.scala

```diff
@@ -1694,6 +1694,9 @@ trait OptimisticTransactionImpl extends TransactionalWrite
         }
         logWarning("Detected no metadata in initial commit but commit validation was turned off.")
       }
+    } else {
+      // Ensure Commit Directory exists if managed commits is enabled on an existing table.
+      newMetadata.flatMap(_.managedCommitOwnerName).foreach(_ => deltaLog.ensureLogDirectoryExist())
```

Comment:
We also need to check that oldMetadata didn't had managed commits? Otherwise we will keep on invoking this function forever for MC tables.

Reply:
We can also move the ensureLogDirectory() logic to a separate block outside this existing if condition.

```
  lazy val isFsToMcConversion = oldMetadata....isEmpty && newMetadata....nonEmpty
  val shouldCreateLogDirs = snapshot.version == -1 || isfsToMcConversion
  if (ishouldCreateLogDirs) {
    deltaLog.ensureLogDirectoryExist()
  }

```

Reply:
> We also need to check that oldMetadata didn't had managed commits? Otherwise we will keep on invoking this function forever for MC tables.

Can we also add a test for this case?

Reply:
Added.